extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/ArcEndpointProcessor.java

```diff
@@ -9,25 +9,27 @@
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
+import io.quarkus.vertx.http.deployment.NonApplicationRootPathBuildItem;
 import io.quarkus.vertx.http.deployment.RouteBuildItem;
-import io.quarkus.vertx.http.runtime.HttpBuildTimeConfig;
 import io.quarkus.vertx.http.runtime.devmode.ArcEndpointRecorder;
 
 public class ArcEndpointProcessor {
 
     @Record(ExecutionTime.RUNTIME_INIT)
     @BuildStep(onlyIf = IsDevelopment.class)
-    void registerRoutes(HttpBuildTimeConfig httpConfig, ArcConfig arcConfig, ArcEndpointRecorder recorder,
+    void registerRoutes(ArcConfig arcConfig, ArcEndpointRecorder recorder,
             BuildProducer<RouteBuildItem> routes,
-            BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints) {
-        String basePath = httpConfig.consolePath + "/arc";
+            BuildProducer<NotFoundPageDisplayableEndpointBuildItem> displayableEndpoints,
+            NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem) {
+        String basePath = nonApplicationRootPathBuildItem.adjustPath("/arc");
```

Comment:
Another way to do this is not adjust the path at this point but call `nonApplicationEndpoint()` on the `RouteItem` builder so that the routes are automatically registered to whatever the non application endpoint path is